emsdk.py

```diff
@@ -1014,15 +1013,48 @@ def xcode_sdk_version():
     return subprocess.checkplatform.mac_ver()[0].split('.')
 
 
+def cmake_target_platform_no_tool_arch(tool):
+    if ARCH == 'aarch64':
+      return 'ARM64'
+    else:
+      return 'x64' if tool.bitness == 64 else 'Win32'
+
+
+def cmake_target_platform(tool):
+  # Source: https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2017%202022.html#platform-selection
+  if hasattr(tool, 'arch'):
+    if tool.arch == 'aarch64':
+      return 'ARM64'
+    elif tool.arch == 'x86_64':
+      return 'x64'
+    elif tool.arch == 'x86':
+      return 'Win32'
+    else:
+      return cmake_target_platform_no_tool_arch(tool)
+  else:
+    return cmake_target_platform_no_tool_arch(tool)
+
+
+def cmake_host_platform():
```

Comment:
Is this a arch breakdown a cmake thing or visual studio thing?  If its the latter maybe call this `vs_host_platform`?

That would also helper reader know this is a windows-only function.

Reply:
AFAIK, it's CMake thing when the generator (`-G`) is `Visual Studio X YYYY` and other generators do not use it but `-A` option is meant to be generic so there might be other uses of it (at least in the future).